api/src/main/java/jakarta/data/metamodel/SortableAttribute.java

```diff
@@ -18,18 +18,17 @@
 package jakarta.data.metamodel;
 
 import jakarta.data.Sort;
+import jakarta.data.metamodel.impl.SortableAttributeRecord;
 
 /**
- * Represents a sortable entity attribute in the {@link StaticMetamodel}.
- * Entity attribute types that are sortable include:
+ * <p>Represents a entity attribute in the {@link StaticMetamodel}
+ * that is sortable, but not comparable.</p>
  *
- * <ul>
- * <li>numeric attributes</li>
- * <li>enum attributes</li>
- * <li>time attributes</li>
- * <li>boolean attributes</li>
- * <li>{@linkplain TextAttribute textual attributes}</li>
- * </ul>
+ * <p>Wherever possible, it is preferable to use {@link ComparableAttribute}
+ * instead because it provides more function. Use {@code SortableAttribute}
+ * only where you have an entity attribute upon which query results can be sorted,
+ * but cannot otherwise be compared within query restrictions. For some databases,
+ * entity attributes of type {@code byte[]} fall under this category.</p>
```

Comment:
```suggestion
 *<p>A {@code SortableAttribute} may be used to sort query results.
 * When an attribute type is a subtype of {@link java.lang.Comparable>, 
 * use of {@link ComparableAttribute} is usually preferred, since a
 * {@code SortableAttribute} may not not used in order-based query 
 * restrictions. Direct use of {@code SortableAttribute} is appropriate 
 * for attributes of type {@code byte[]}.</p>
```

Reply:
This has one minor type, which I'll correct later, but otherwise looks good